packages/myst-frontmatter/src/utils/parseName.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { describe, expect, test } from 'vitest';
+import type { Name } from '../frontmatter/types.js';
+import { parseName, renderName, startsWithUpperCase } from './parseName.js';
+
+describe('Test name parsing', () => {
+  test.each([
+    ['Abc', true],
+    ['ABC', true],
+    ['aBC', false],
+    ['abc', false],
+    ['àbc', false],
+    ['àBC', false],
+    ['Àbc', true],
+    ['1Abc', true],
+    ['1aBC', false],
+    ['123', true],
+  ])('test startsWithUpperCase: %s', async (word, upper) => {
+    expect(startsWithUpperCase(word)).toEqual(upper);
+  });
+  test.each<[string, Name, string?]>([
+    ['aa', { display: 'aa', family: 'aa' }, 'aa,'],
```

Comment:
it would be good to get some real names in the tests 😬 these character chucks are fine mechanically but it's difficult to see the need for a dropped particle in these examples

Reply:
here's an interesting test set thand to genai :p I think we want to limit to english and I don't think we can ignore dropped particle

```
* Simple Given and Family Name:

  * "John Doe"
  * "Alice Smith"
  * "Luis Garcia"

* Single Name:

  * "Madonna"
  * "Cher"
  * "Prince"

* Compound Given Names:

  * "Mary Jane Watson"
  * "Billy Bob Thornton"
  * "Jean Paul Gaultier"


* Particles (like 'de', 'von', 'van', 'di' often found in European names):

  * "Vincent van Gogh" (Given: Vincent, Particle: van, Family: Gogh)
  * "Ludwig von Beethoven" (Given: Ludwig, Particle: von, Family: Beethoven)
  * "Leonardo di Caprio" (Given: Leonardo, Particle: di, Family: Caprio)
  * "João de Silva" (Given: João, Particle: de, Family: Silva)

* Dropped Particles (which might not always be included, like 'le' or 'la'):

  * "Charles le Brun" or "Charles Brun" (Given: Charles, Dropped Particle: le, Family: Brun)
  * "Isabelle la Catholique" or "Isabelle Catholique" (Given: Isabelle, Dropped Particle: la, Family: Catholique)

Hyphenated Names:

  * "Jamie-Lee Curtis"
  * "Anne-Marie Duff"
  * "Jean-Luc Picard"

* Cultural Variants and Exceptions:

  * "Kim Jong Un" (In Korean names, 'Kim' is the family name)
  * "Xi Jinping" (In Chinese names, 'Xi' is the family name)
  * "Mohammed bin Salman" (Arabic, where 'bin' signifies 'son of')

* Names with Apostrophes or Other Punctuation:

  * "D'Arcy Wretzky"
  * "O'Connor"
  * "L'Engle"
  * "Jean-François"

* Titles and Suffixes (not asked for, but they can complicate parsing):

  * "Dr. Jane Doe"
  * "Sir Paul McCartney"
  * "John Smith Jr."

* Initials:

  * "J.K. Rowling"
  * "T.S. Eliot"
  * "F. Scott Fitzgerald"
```